Source/WebCore/bindings/js/JSDOMWindowBase.cpp

```diff
@@ -315,7 +315,11 @@ JSValue toJS(JSGlobalObject* lexicalGlobalObject, DOMWindow& domWindow)
 
 JSDOMWindow* toJSDOMWindow(Frame& frame, DOMWrapperWorld& world)
 {
-    return frame.script().globalObject(world);
+    if (auto* window = JSC::jsDynamicCast<JSDOMWindow*>(frame.script().globalObject(world))) {
```

Comment:
In which case isn't the global object of a Frame a `JSDOMWindow*`?

I have concerns this shouldn't be possible and may negatively impact performance.

Note that remote windows should have remote frames. Here we have a (Local)Frame so I don't really understand why its global object wouldn't be a JSDOMWindow.